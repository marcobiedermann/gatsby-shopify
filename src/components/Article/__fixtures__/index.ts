import { ArticleProps } from '..';

const article: ArticleProps = {
  blog: {
    handle: 'news',
  },
  excerptHtml: '',
  handle: 'article-title',
  shopifyId: '12345678-9012-3456-7890-123456789012',
  title: 'Article Title',
};

export default article;
