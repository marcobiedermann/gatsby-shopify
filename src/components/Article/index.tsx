import { Link } from 'gatsby';
import React, { FC } from 'react';

export interface ArticleProps {
  blog: {
    handle: string;
  };
  handle: string;
  shopifyId: string;
  title: string;
}

const Article: FC<ArticleProps> = (props) => {
  const { blog, handle, title } = props;

  return (
    <article>
      <h3>
        <Link to={`/blogs/${blog.handle}/${handle}`}>{title}</Link>
      </h3>
    </article>
  );
};

export default Article;
