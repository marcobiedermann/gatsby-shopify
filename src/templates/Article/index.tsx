import { graphql, PageProps } from "gatsby"
import React, { FC } from "react"
import Layout from "../../components/Layout"

interface DataProps {
  shopifyArticle: {
    author: {
      name: string
    }
    contentHtml: string
    publishedAt: string
    title: string
  }
}

const ArticleTemplate: FC<PageProps<DataProps>> = props => {
  const {
    data: { shopifyArticle },
  } = props

  return (
    <Layout>
      <h1>{shopifyArticle.title}</h1>
      <div dangerouslySetInnerHTML={{ __html: shopifyArticle.contentHtml }} />
    </Layout>
  )
}

export const pageQuery = graphql`
  query ArticleByShopifyId($shopifyId: String!) {
    shopifyArticle(shopifyId: { eq: $shopifyId }) {
      author {
        name
      }
      contentHtml
      publishedAt
      title
    }
  }
`

export default ArticleTemplate
