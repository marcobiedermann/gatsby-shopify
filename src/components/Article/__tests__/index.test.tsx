import { render } from '@testing-library/react';
import React from 'react';
import Article from '..';
import article from '../__fixtures__';

describe('article component', () => {
  it('should render correctly', () => {
    const { container } = render(<Article {...article} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
