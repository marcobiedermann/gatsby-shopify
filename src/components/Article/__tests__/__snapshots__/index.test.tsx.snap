// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article component should render correctly 1`] = `
<article>
  <h3>
    <a
      href="/blogs/news/article-title"
    >
      Article Title
    </a>
  </h3>
  <div />
  <p>
    <a
      href="/blogs/news/article-title"
    >
      Read more
    </a>
  </p>
</article>
`;
